source/gui/exit.py

```diff
@@ -0,0 +1,163 @@
+# A part of NonVisual Desktop Access (NVDA)
+# Copyright (C) 2006-2022 NV Access Limited, Peter Vágner, Aleksey Sadovoy, Mesar Hameed, Joseph Lee,
+# Thomas Stivers, Babbage B.V., Accessolutions, Julien Cochuyt, Cyrille Bougot
+# This file is covered by the GNU General Public License.
+# See the file COPYING for more details.
+
+
+import config
+import core
+from enum import auto, unique
+import globalVars
+import languageHandler
+from logHandler import log
+import queueHandler
+from utils.displayString import DisplayStringEnum
+import weakref
+import wx
+
+from . import guiHelper
+from .startupDialogs import WelcomeDialog
+
+
+try:
+	import updateCheck
+except RuntimeError:
+	updateCheck = None
+
+
+@unique
+class _ExitAction(DisplayStringEnum):
+
+	EXIT = auto()
+	RESTART = auto()
+	RESTART_WITH_ADDONS_DISABLED = auto()
+	RESTART_WITH_DEBUG_LOGGING_ENABLED = auto()
+	INSTALL_PENDING_UPDATE = auto()
+
+	@property
+	def _displayStringLabels(self):
+		return {
+			# Translators: An option in the combo box to choose exit action.
+			self.EXIT: _("Exit"),
+			# Translators: An option in the combo box to choose exit action.
+			self.RESTART: _("Restart"),
+			# Translators: An option in the combo box to choose exit action.
+			self.RESTART_WITH_ADDONS_DISABLED: _("Restart with add-ons disabled"),
+			# Translators: An option in the combo box to choose exit action.
+			self.RESTART_WITH_DEBUG_LOGGING_ENABLED: _("Restart with debug logging enabled"),
+			# Translators: An option in the combo box to choose exit action.
+			self.INSTALL_PENDING_UPDATE: _("Install pending update"),
+		}
+
+
+def quit():
```

Comment:
I'm not really happy wit how this function is named / what it is doing. This is not a new thing - the previous `gui.quit` was also rather confusing but since   we're in the process of refactoring this code anyway perhaps we can try to make this clearer.
Areas which should be considered for improvements:
- naming: `gui.quit` suggests to the reader that this function quits the graphical interface whereas it just calls an event on the main frame which either shows exit dialog or closes the entire screen reader depending on the user preference. The new placement `gui.exit.quit` is even worse IMO as it suggests this function is responsible for quitting the exit dialog.
- Usage - why we need a separate function for invoking an event on a main frame at all.

At a minimum this function should get a docstring to explain  what it is doing. Even better would be to make it a class method on exitdialog (perhaps something along the lines of `showExitOptionsOrQuit`) and modify it so that it either shows the dialog if configured or just exist NVDA.

Reply:
@feerrenrut 
Based on this comment, I'd like to update this PR to mark `gui.quit` for deprecation.
It's only used once in NVDA core, and is a redundant helper function.
Note that the other global commands use the pattern of `wx.CallAfter(gui.mainFrame.onExampleCommand)`.

I'll have to rebase this to perform the patch, so I'd like approval first before doing so.

The patch I'm planning to perform:

```diff
diff --git a/source/globalCommands.py b/source/globalCommands.py
index 46a94c0d7..e9f301f9e 100755
--- a/source/globalCommands.py
+++ b/source/globalCommands.py
@@ -1757,7 +1757,7 @@ class GlobalCommands(ScriptableObject):
 		gesture="kb:NVDA+q"
 	)
 	def script_quit(self,gesture):
-		gui.exit.quit()
+		wx.CallAfter(gui.mainFrame.onExitCommand, None)
 
 	@script(
 		# Translators: Input help mode message for restart NVDA command.
diff --git a/source/gui/__init__.py b/source/gui/__init__.py
index 036f7d6a4..db6ffc313 100644
--- a/source/gui/__init__.py
+++ b/source/gui/__init__.py
@@ -50,11 +50,13 @@ from buildVersion import version_year
 
 
 if version_year < 2023:
-	from .exit import quit  # noqa F401: imported for backward compatibility
-	"""
-	Deprecated, for removal in 2023.
-	Use L{gui.exit.quit} instead.
-	"""
+	def quit():
+		"""
+		Deprecated, for removal in 2023.
+		Use `wx.CallAfter(mainFrame.onExitCommand, None)` directly instead.
+		"""
+		wx.CallAfter(mainFrame.onExitCommand, None)
+
 
 
 try:
diff --git a/source/gui/exit.py b/source/gui/exit.py
index eebd42449..7081c6535 100644
--- a/source/gui/exit.py
+++ b/source/gui/exit.py
@@ -51,11 +51,6 @@ class _ExitAction(DisplayStringEnum):
 		}
 
 
-def quit():
-	from gui import mainFrame
-	wx.CallAfter(mainFrame.onExitCommand, None)
-
-
 class ExitDialog(wx.Dialog):
 	_instance = None
```